src/Algebra/Graph/AdjacencyIntMap/Algorithm.hs

```diff
@@ -115,6 +118,56 @@ dfs vs = concatMap flatten . dfsForestFrom vs
 reachable :: Int -> AdjacencyIntMap -> [Int]
 reachable x = dfs [x]
 
+-- | Compute the forest of a graph's vertices in breadth first order.
+-- Complexity: /O(n+m*log n)/ time and /O(n+m)/ space.
+--
+-- @
+-- bfsForest 'empty'                         == []
+-- 'forest' (bfsForest $ 'edge' 1 1)           == 'vertex' 1
+-- 'forest' (bfsForest $ 'edge' 1 2)           == 'edge' 1 2
+-- 'forest' (bfsForest $ 'edge' 2 1)           == 'vertices' [1,2]
+-- 'isSubgraphOf' ('forest' $ bfsForest x) x   == True
+-- bfsForest . 'forest' . bfsForest          == bfsForest
+-- 'forest' (bfsForest ('circuit' [1..5] + 'transpose' ('circuit' [1..5]))) == 'path' [1,2,3] + 'path' [1,5,4]
+-- @
+bfsForest :: AdjacencyIntMap -> Forest Int
+bfsForest g = bfsForestFrom' (vertexList g) g
+
+-- | Like 'bfsForest', but the traversal is seeded by a list of
+-- vertices.  Let /L/ be the number of seed vertices. Complexity:
+-- /O(n+(L+m)*log n)/ time and /O(n+m)/ space.
+--
+-- @
+-- 'forest' (bfsForestFrom [1,2] $ 'edge' 1 2) == 'vertices' [1,2]
+-- 'forest' (bfsForestFrom [2]   $ 'edge' 2 1) == 'vertex' 2
```

Comment:
I don't think the examples here work, and they seem different from tests. Please bring them in sync.

Reply:
Yeah, thankfully the test cases were what they ought to be, and passed! I'm not sure how I got the documentation examples so wrong

Reply:
fixed to `forest (bfsForestFrom [2]   $ edge 1 2) == vertex 2`